### PR TITLE
scaffolder-backend: Allow more options in publish:github action

### PR DESCRIPTION
**Pull request by peimanja**
## Hey, I just made a Pull Request!

Allow passing the following options in `publish:github` action.
- delete_branch_on_merge
- allow_squash_merge
- allow_merge_commit
- allow_rebase_merge

Keeping the defaults to not introduce any breaking changes 

**Motivation:**

When creating a repository, we have `delete_branch_on_merge` set to `true` and only allow `squash merge` in our org and would like to keep it that way when using Backstage software templates

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
Makes sense, thank you! 👍

**Comment by peimanja** [comment]
@Rugvip Thanks for the quick review. I had to update `mockContext` since CI checks were failing

**Review by benjdlambert — COMMENTED** [review]
Thanks for this 👍  Just a comment about the options being optional.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ export function createPublishGithubAction(options: {
     description?: string;
     access?: string;
     defaultBranch?: string;
+    deleteBranchOnMerge: boolean;
```
[the remark this answers is not in the document]
Comment (peimanja) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/github.ts:49]: sure makes sense

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**